magicctapipe/io/tests/test_io.py

```diff
@@ -19,619 +21,721 @@
-    ):
-        check_input_list(config_check)
-
-    with pytest.raises(
-        Exception,
-        match="Number of telescopes found in the configuration file is 5. It must be 6, i.e.: LST-1, LST-2, LST-3, LST-4, MAGIC-I, and MAGIC-II.",
-    ):
         check_input_list(
             {
                 "mc_tel_ids": {
                     "LST-1": 1,
                     "LST-2": 2,
                     "LST-3": 3,
-                    "MAGIC-I": 4,
-                    "MAGIC-II": 5,
+                    "LST-4": 4,
+                    "MAGIC-I": 5,
+                    "MAGIC-II": 6,
                 }
             }
         )
 
-    with pytest.raises(
-        Exception,
-        match="Number of telescopes found in the configuration file is 7. It must be 6, i.e.: LST-1, LST-2, LST-3, LST-4, MAGIC-I, and MAGIC-II.",
-    ):
         check_input_list(
             {
                 "mc_tel_ids": {
                     "LST-1": 1,
-                    "LST-2": 2,
-                    "LST-3": 3,
-                    "LST-4": 6,
-                    "LST-5": 7,
-                    "MAGIC-I": 4,
-                    "MAGIC-II": 5,
+                    "LST-2": 3,
+                    "LST-3": 0,
+                    "LST-4": 0,
+                    "MAGIC-I": 2,
+                    "MAGIC-II": 6,
                 }
             }
         )
 
-    with pytest.raises(
-        Exception,
-        match="Entry 'LSTT-1' not accepted as an LST. Please make sure that the first four telescopes are LSTs, e.g.: 'LST-1', 'LST-2', 'LST-3', and 'LST-4'",
-    ):
         check_input_list(
             {
                 "mc_tel_ids": {
-                    "LSTT-1": 1,
-                    "LST-2": 2,
-                    "LST-3": 3,
-                    "LST-4": 6,
-                    "MAGIC-I": 4,
-                    "MAGIC-II": 5,
+                    "LST-2": 1,
+                    "LST-1": 3,
+                    "LST-4": 0,
+                    "LST-3": 0,
+                    "MAGIC-II": 2,
+                    "MAGIC-I": 6,
                 }
             }
         )
 
-    with pytest.raises(
-        Exception,
-        match="Entry 'MAGIC-III' not accepted as a MAGIC. Please make sure that the last two telescopes are MAGICs, e.g.: 'MAGIC-I', and 'MAGIC-II'",
-    ):
-        check_input_list(
-            {
-                "mc_tel_ids": {
-                    "LST-1": 1,
-                    "LST-2": 2,
-                    "LST-3": 3,
-                    "LST-4": 6,
-                    "MAGIC-I": 4,
-                    "MAGIC-III": 5,
-                }
-            }
-        )
+        with pytest.raises(
+            Exception,
+            match="Number of telescopes found in the configuration file is 5. It must be 6, i.e.: LST-1, LST-2, LST-3, LST-4, MAGIC-I, and MAGIC-II.",
+        ):
+            check_input_list(config_check)
```

Comment:
Oh yes I remember why I did this. This test check what can happen if we make a typo in the config file so that we have two tels with the same name (eg. LST-2 and LST-2 instead of LST-1 and LST-2). If you hard-code this kind of combination into the test script, the python interpreter will complain because this is a Python error; if you put into the yaml file you can verify if an exception is raised (yes, it is raised)

Reply:
I think it must depend on the actual version of interpretter, because in ipython I get this:
```
In [1]: d1 = {"a": 1, "b": 2, "a": 3}

In [2]: d1
Out[2]: {'a': 3, 'b': 2}
```

A dictionary cannot have two same keys, so what is probably happening is that it is reading the same key twice and updating it. If you print the dictionary I'm pretty sure you will have only once LST-2, and the test triggers not because there are two entries with the same name, but because there is one missing.

Reply:
I removed this section of the test because if we hard-code the dictionary pyflakes checks will fail. BTW, this was not a very important test